pluginutils (via TypeScript plugin): Cannot find module 'estree'

<!--
  ⚡️ katchow! We 💛 issues.

  🚨 Your issue will be CLOSED if:
     - This template is removed
     - Parts of this template are removed

  👉🏽 Need help or tech support? Please don't open an issue!
  Head to https://is.gd/rollup_chat or https://stackoverflow.com/questions/tagged/rollupjs

  ❤️ Rollup? Please consider supporting our collective:
  👉 https://opencollective.com/rollup/donate
-->

- Rollup Plugin Name: `@rollup/plugin-typescript`
- Rollup Plugin Version: 9.0.1
- Rollup Version: 3.1.0
- Operating System (or Browser): Windows
- Node Version: 18.10.0
- Link to reproduction: https://github.com/meyfa/rollup-typescript-estree-repro

<!--
  🚨 Issues WITHOUT a valid reproduction WILL BE CLOSED!

  Please provide one by:
  1. Using the REPL.it plugin reproduction template at https://repl.it/@rollup/rollup-plugin-repro
  2. Provide a minimal repository link (Read https://gist.github.com/Rich-Harris/88c5fc2ac6dc941b22e7996af05d70ff for instructions).
     Please use NPM for installing dependencies!
     These may take more time to triage than the other options.
  3. Using the Rollup REPL at https://rollupjs.org/repl/

  ⚠️ ZIP Files are unsafe and maintainers will NOT download them.
-->

I'm getting an error with the newest version of the TypeScript plugin. This did not happen before the Rollup 3 preparations.

How to reproduce:

1. `npm install --save-dev typescript rollup @rollup/plugin-typescript`
2.  Define a config with the typescript plugin.
3. Perform a TypeScript typecheck, e.g. via `tsc --noEmit`.


### Expected Behavior

The typecheck succeeds.

### Actual Behavior

TypeScript outputs the following error:

```
node_modules/@rollup/pluginutils/types/index.d.ts:2:26 - error TS2307: Cannot find module 'estree' or its corresponding type declarations.

2 import { BaseNode } from 'estree';
                           ~~~~~~~~

Found 1 error in node_modules/@rollup/pluginutils/types/index.d.ts:2
```

### Additional Information

<!--
  Most issues can be expressed or demonstrated through the REPL or a repository.
  However, the situation may arise where some small code snippets also need to
  be provided. In that situation, please add your code below using
  Fenced Code Blocks (https://help.github.com/articles/creating-and-highlighting-code-blocks/)
-->

* `@rollup/plugin-typescript` still depends on `@rollup/pluginutils` version `^4.2.1`, while `5.0.0` would be available. Maybe this is part of the issue?
* I looked at some other issues with this error, especially #1156, but they were never resolved.

--- a/packages/pluginutils/package.json
+++ b/packages/pluginutils/package.json
@@ -61,6 +61,7 @@
     }
   },
   "dependencies": {
+    "@types/estree": "^1.0.0",
     "estree-walker": "^2.0.2",
     "picomatch": "^2.3.1"
   },
@@ -68,7 +69,6 @@
     "@rollup/plugin-commonjs": "^22.0.2",
     "@rollup/plugin-node-resolve": "^14.1.0",
     "@rollup/plugin-typescript": "^8.5.0",
-    "@types/estree": "1.0.0",
     "@types/node": "^14.18.30",
     "@types/picomatch": "^2.3.0",
     "acorn": "^8.8.0",

